utils/installer/install-neovim-from-release

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+set -eu pipefall
+
+declare -r INSTALL_PREFIX="${INSTALL_PREFIX:-"$HOME/.local"}"
+declare -r RELEASE_VER="${RELEASE_VER:-latest}" # can be set to nightly
+
+declare ARCHIVE_NAME
+declare RELEASE_NAME
+declare OS
+
+OS="$(uname -s)"
+
+if [ "$OS" == "Linux" ]; then
+  ARCHIVE_NAME="nvim-linux64"
+  RELEASE_NAME="nvim-linux64"
+elif [ "$OS" == "Darwin" ]; then
+  ARCHIVE_NAME="nvim-macos"
+  # for some reason the archive has a different name
+  RELEASE_NAME="nvim-osx64"
+else
+  echo "$OS platform is not supported currently"
+  exit 1
+fi
+
+declare -r RELEASE_URL="https://github.com/neovim/neovim/releases/$RELEASE_VER/download/$ARCHIVE_NAME.tar.gz"
+declare -r CHECKSUM_URL="$RELEASE_URL.sha256sum"
+
+DOWNLOAD_DIR="$(mktemp -d)"
+readonly DOWNLOAD_DIR
+
+RELEASE_SHA="$(curl -Ls "$CHECKSUM_URL" | awk '{print $1}')"
+readonly RELEASE_SHA
+
+function main() {
+  if [ ! -d "$INSTALL_PREFIX" ]; then
+    mkdir -p "INSTALL_PREFIX" || __invalid__prefix__handler
+  fi
+  download_neovim
+  verify_neovim
+  install_neovim
+}
+
+function download_neovim() {
+  echo "Downloading Neovim's binary from $RELEASE_VER release.."
+  if ! curl --progress-bar --fail -L "$RELEASE_URL" -o "$DOWNLOAD_DIR/$ARCHIVE_NAME.tar.gz"; then
+    echo "Download failed.  Check that the release/filename are correct."
+    exit 1
+  fi
+  echo "Download complete!"
+}
+
+function verify_neovim() {
+  echo "Verifying the installation.."
+  DOWNLOADED_SHA="$(sha256sum "$DOWNLOAD_DIR/$ARCHIVE_NAME.tar.gz" | awk '{print $1}')"
+
+  if [ "$RELEASE_SHA" != "$DOWNLOADED_SHA" ]; then
```

Comment:
You can use `sha256sum --check $RELEASE_SHA $DOWNLOADED_SHA` to verify

Reply:
Thanks for the tip! I still want to print the  `$DOWNLOADED_SHA` in case of an error, so I'll keep this way for now.